cvmfs/upload_s3.cc

```diff
@@ -361,11 +361,14 @@ void S3Uploader::FileUpload(
                             const_cast<void*>(
                                 static_cast<void const*>(callback)),
                             local_path);
+
+  if (remote_path.substr(0, 1) == ".") {
```

Comment:
Probably works but would also match "./data/<hash>".  Perhaps it's better to check for ".cvmfs" here